### DISCUSSION TITLE
Does the existing PYG framework supports the imbalanced graph size in graph classification task?

### DISCUSSION THREAD

**Discussion by lichen0411** [pyg-team/pytorch_geometric#5995] [Q&A]
I recently encountered a problem with graph classification. For the application I am researching now, the extracted key-points are designed to construct graphs. However, the number of graph nodes created by different samples is different. Specifically, the graph size of the intra-class may have a large difference and the graph size of the inter-class may have a small difference. I tried different graph classification methods such as GCN, GAT, diffpool, and some dynamic graph convolution methods such as AdaptConv, DGCNN, etc. However, their performance is less satisfactory. I also tried some graph similarity learning methods such as MGMN using a cross-attention to learn the cross-graph features and found the optimization is hard resulting in a bad performance. I guess the problem may be the constructed graph is insufficient to represent to target, or the existing graph classification methods and graph similarity learning methods are sensitive to the imbalanced graph size. I'm not confident about whether my guess is right. So does the existing PYG framework supports the imbalanced graph size in graph classification task? Sincerely looking for your response.

**Comment by rusty1s** [comment]
All PyG models can naturally be applied on graphs with varying size. There are multiple ways to either ignore or respect graph sizes during classification tasks (e.g., `global_mean_pool` vs `global_max_pool`, or making use of `GraphSizeNorm`). Hope that helps.

**Reply by lichen0411** [reply]
Sincerely appreciate your answer. I understand that using pooling methods or any readout operation can cope with graphs with varying sizes, while I found in my applications the graph sizes varies greatly, and using global_mean_pool or global_max_pool does not efficiently improve the graph classification performance. So I would like to know whether there're some graph classification methods specifically designed for varying graph sizes. If the graphs I constructed are not good enough, are there some simple methods to evaluate the quality of constructed graphs to keep less intra-class distance and greater inter-class distance?

**Reply by rusty1s** [reply]
Mh, off the top of my head, I cannot think of a GNN that is designed for varying graph sizes in mind. If different graph sizes appear in your training dataset, then the model should be able to either ignore or take that information into account. 

> simple methods to evaluate the quality of constructed graphs to keep less intra-class distance and greater inter-class distance?

You mean like homophily ratio? Take a look at `utils.homophily`.
